### ISSUE TITLE
Add support for 4-part base version

### ISSUE THREAD

**Issue by ImrePyhvel** [notpeelz/VersionInfoGenerator#31]
## Context 
Semantic versioning and it's 3-part format is good and in place for public libraries, etc, but .Net historically works on versions with 4 parts:
```<major>.<minor>.<build>.<revision>```
Obviously actual semantics may vary, but assembly loading, `System.Version` class, `Assembly.GetExecutingAssembly().GetName().Version` and many other tools are still working on this 4-part version numbering scheme, as well as a lot of legacy applications. And will continue to do that for a long time due to existing versioning processes or backward compatibility. VersionInfoGenerator should support this as a compatible, fairly common scenario.

## Repro
VersionInfoGenerator 3.0.0 + new .Net8 project.
Set up base version in csproj file a la:
```
<Version>12.1.3.8</Version>
```

## Actual behavior:
VersionInfoGenerator outright rejects entire build with message:
```
Error	VIG1100	Version does not respect the Major.Minor.Patch format: 12.1.3.8
```
Requiring to ditch VersionInfoGenerator entirely.

## Expected behavior:
VersionInfo should accept the historically valid 4-part version number for build, and parse its `major`, `minor`, `patch` from first 3 parts and ignore the rest.
Build should succeed, git info and the rest should still be usable in project.

### Alternative/optional additon: support configuring custom base version
If accepting 4-part version by default is not acceptable, then it would suffice to provide an configuration option on project MSBuild to explicitly provide custom version number (or major+minor+patch).

This extension point would also open up fetching the base version from outside the csproj, i.e. from text file used across all projects in solution, which is quite often used in CI scenarios.

**Comment by notpeelz** [comment]
I originally considered implementing version normalization [à la NuGet](https://learn.microsoft.com/en-us/nuget/concepts/package-versioning#normalized-version-numbers), but decided against it since I had no interest in supporting the legacy version format.

I could add a `VersionValidate` setting to disable this check, but the "SemVer" string technically won't be a semver anymore.
Would that work for you?

**Comment by ImrePyhvel** [comment]
Nuget-like normalization looks very reasonable to me, but I do fully understand the desire of NOT having a legacy baggage tagging along. 

Simple option to just turn off the validation would suffice for me, as long as the following logic will not choke on 4-part version number input. As it would be an opt-in feature, then it would be a non-breaking change.

Regarding change in semantics of generated `SemVer` field, when having a version number not adhering to semVer base version number format and semantics to start with, then the change in output semantics seems obvious. I don't think that's a problem, the generated parts are still meaningful and useful, just not to be interpreted using SemVer standard.

**Comment by notpeelz** [comment]
Implemented VersionValidate in 3.1.0
